CMakeLists.txt

```diff
@@ -0,0 +1,4 @@
+# This file is here in the root folder so it can be opened as a CMake project in Visual Studio, eg in order to debug a Linux build in WSL2.
+cmake_minimum_required(VERSION 3.13)
+project(TryWSL)
```

Comment:
Using cmake was a bit of an experiment I started when I joined, so the main CMakeLists.txt started of in that contrib/cmake folder.   If we get agreement from the team, perhaps we could move that here, but in the meantime this is sort of a pointer to that folder.    That particular 'project' line was needed to prevent cmake complaining - I think the later 'project' statement in the real CMakeLists.txt overwrites it anyway.   And that one can be user-specified through cache variables, I use that so that when I have multiple Visual Studios open, the project name reminds me what I'm working on in each one.